diff --git a/WheelWizard/Services/Launcher/Helpers/DolphinLaunchHelper.cs b/WheelWizard/Services/Launcher/Helpers/DolphinLaunchHelper.cs
index 12dca25a..ac2135d2 100644
--- a/WheelWizard/Services/Launcher/Helpers/DolphinLaunchHelper.cs
+++ b/WheelWizard/Services/Launcher/Helpers/DolphinLaunchHelper.cs
@@ -116,7 +116,9 @@ public static class DolphinLaunchHelper
 
         if (!TryFixFlatpakPortalAccess(PathManager.GameFilePath, "-r"))
             AddFilesystemPerm(PathManager.GameFilePath, ":ro");
-        AddFilesystemPerm(PathManager.RrLaunchJsonFilePath, ":ro");
+        // We need to provide the directory where the `RR.json` is located in for portal access!
+        if (!TryFixFlatpakPortalAccess(Path.GetDirectoryName(PathManager.RrLaunchJsonFilePath) ?? "", "-r"))
+            AddFilesystemPerm(PathManager.RrLaunchJsonFilePath, ":ro");
 
         return fixedFlatpakDolphinLocation;
     }
