diff --git a/Directory.Build.props b/Directory.Build.props
index b02e2e6e..1dcbc498 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -54,5 +54,6 @@
             <PrivateAssets>all</PrivateAssets>
             <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
         </PackageReference>
+        <PackageReference Include="DotNet.ReproducibleBuilds" Version="2.0.2" PrivateAssets="All"/>
     </ItemGroup>
-</Project>
\ No newline at end of file
+</Project>
